CMakeLists.txt

```diff
@@ -224,19 +224,23 @@ if(USE_CUDA)
     # reference for mapping of CUDA toolkit component versions to supported architectures ("compute capabilities"):
     # https://en.wikipedia.org/wiki/CUDA#GPUs_supported
     set(CUDA_ARCHS "60" "61" "62" "70" "75")
-    if(CUDA_VERSION VERSION_GREATER_EQUAL "110")
+    if(CUDAToolkit_VERSION VERSION_GREATER_EQUAL "11.0")
```

Comment:
As we call [FindCUDAToolkit](https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html) but not [FindCUDA](https://cmake.org/cmake/help/latest/module/FindCUDA.html), we get undefined `CUDA_VERSION` variable.
https://github.com/microsoft/LightGBM/blob/425395de09a65a37bfc3eea48b509cec86642b4b/CMakeLists.txt#L220

Reply:
Excellent fix, thank you!